### PR TITLE
fvm debug execution

### PR DESCRIPTION
**Pull request by vyzo**
Companion to https://github.com/filecoin-project/ref-fvm/pull/605

### CODE REVIEW

File: rust/src/fvm/machine.rs [hunk 1]
```diff
@@ -132,6 +132,84 @@ fn create_fvm_machine(
     }
 }
 
+#[ffi_export]
+fn create_fvm_debug_machine(
```
Comment (raulk) [inline rust/src/fvm/machine.rs:136]: Can we refactor this to avoid duplicating the whole function?

Comment (vyzo) [reply rust/src/fvm/machine.rs:136]: sure, once it provably works :)

**Comment by raulk** [comment]
@vyzo Can you please remove the duplication so we can merge this PR?

**Comment by vyzo** [comment]
yes of course!

File: rust/src/fvm/machine.rs (hunk 1 above)
Comment (vyzo) [reply rust/src/fvm/machine.rs:192]: done.

**Comment by vyzo** [comment]
Changes:
- rebased on master and squashed a few commits
- patched ref-fvm to point to `release/v1` branch which contains the debug execution feature
- deduplicated the code.

Should be ready now.

**Review by raulk — COMMENTED** [review]
LGTM besides the comments. Would prefer @Stebalien to give the final signoff as a more active maintainer of this repo.

File: rust/src/fvm/machine.rs [hunk 2]
```diff
@@ -80,19 +78,20 @@ fn create_fvm_machine(
...
+                }
                 // handle cid.Undef for no manifest
                 // this can mean two things:
                 // - for pre nv16, use the builtin bundles
                 // - for nv16 or higher, it means we have already migrated state for system
                 //   actor and we can pass None to the machine constructor to fish it from state.
                 // The presence of the manifest cid argument allows us to test with new bundles
                 // with minimum friction.
-                None
+                None => None,
```
[the remark this answers is not in the document]
Comment (vyzo) [reply rust/src/fvm/machine.rs:94]: map didnt like me, but i can temove the comment completely.

File: rust/src/fvm/machine.rs [hunk 3]
```diff
@@ -102,9 +101,28 @@ fn create_fvm_machine(
                 Ok(Some(manifest)) => {
                     network_config.override_actors(manifest);
                 }
-                Ok(None) => {}
+                Ok(None) => (),
```
[the remark this answers is not in the document]
Comment (vyzo) [reply rust/src/fvm/machine.rs:104]: its equivalent, so its a matter of style; fine with reverting it.

**Review by Stebalien — APPROVED** [review]
This works (once we've removed the branch dependencies). Are we planning on getting this into the lotus release? If so, we should revert the dependency updates in Cargo.toml.

**Comment by vyzo** [comment]
I think we are aiming for lotus master -- @arajasek ?

File: rust/src/fvm/machine.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vyzo) [reply rust/src/fvm/machine.rs]: it came from outer space... this language is an obscurity competition.

**Pull request merged**